applications/asset_tracker/src/main.c

```diff
@@ -812,6 +860,42 @@ static void app_connect(struct k_work *work)
 	}
 }
 
+/**@brief Disconnects from cloud. First it tries using the cloud backend's
+ *	  disconnect() implementation. If that fails, it falls back to close the
+ *	  socket directly, using close().
+ */
+static void app_disconnect(void)
+{
+	int err;
+
+	atomic_set(&send_data_enable, 0);
+	printk("Disconnecting from cloud.\n");
+
+	err = cloud_disconnect(cloud_backend);
+	if (err == -ENOTCONN) {
+		printk("Cloud connection was not established.\n");
+		return;
+	}
+
+	if (err) {
+		printk("Could not disconnect from cloud, err: %d\n", err);
+		printk("Closing the cloud socket directly\n");
+
+		err = close(cloud_backend->config->socket);
+		if (err) {
+			printk("Failed to close socket, error: %d\n", err);
+			return;
+		}
+
+		printk("Socket was closed successfully\n");
+	}
+
+	/* Ensure that the socket is indeed closed before returning. */
+	k_sem_take(&cloud_disconnected, K_FOREVER);
+
+	printk("Controlled disconnect from cloud successful.\n");
```

Comment:
`Controlled disconnect` that sounds odd, what did you mean here?

Reply:
Yep, that is weird, changed to `Disconnected from cloud`.